solr/core/src/java/org/apache/solr/cluster/placement/plugins/AffinityPlacementConfig.java

```diff
@@ -43,14 +46,28 @@
   @JsonProperty
   public long prioritizedFreeDiskGB;
 
+  /**
+   * This property defines an additional constraint that primary collections (keys) should be
+   * located on the same nodes as the secondary collections (values). The plugin will assume
+   * that the secondary collection replicas are already in place and ignore candidate nodes where
+   * they are not already present.
+   */
+  @JsonProperty
+  public Map<String, String> withCollections;
+
   // no-arg public constructor required for deserialization
   public AffinityPlacementConfig() {
-    minimalFreeDiskGB = 20L;
-    prioritizedFreeDiskGB = 100L;
+    this (20L, 100L, Map.of());
```

Comment:
minor: remove space after `this`.

Reply:
Also, this constructor should call the constructor not requiring a `Map`, it will add an empty one.

BTW, the deserialization constructor should not pass "meaningful" default values since those will not be used.
I'd call `this(0L, 0L)` here instead, and switch the `AffinityPlacementConfig DEFAULT` static variable to use the constructor accepting values. That will also make seeing what are the defaults easier when reading the code.